sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -3740,3 +3743,147 @@ def hpat_pandas_series_pct_change_impl(self, periods=1, fill_method='pad', limit
+                sorted_percentiles = sorted(percentiles_list)
+
+                # check percentiles have correct values:
+                arr = numpy.asarray(sorted_percentiles)
+                if len(numpy.unique(arr)) != len(arr):
+                    raise ValueError("percentiles cannot contain duplicates")
+                if numpy.any(arr[(arr < 0) * (arr > 1)]):
+                    raise ValueError("percentiles should all be in the interval [0, 1].")
+
+                # TODO: support proper rounding of percentiles like in pandas.io.formats.format.format_percentiles
+                # requires numpy.round(precision), numpy.isclose to be supported by Numba
+                percentiles_indexes = []
+                for percentile in arr:
+                    p_as_string = str(percentile * 100)
+
+                    trim_index = len(p_as_string) - 1
+                    while trim_index >= 0:
+                        if p_as_string[trim_index] == '0':
+                            trim_index -= 1
+                            continue
+                        elif p_as_string[trim_index] == '.':
+                            break
+
+                        trim_index += 1
+                        break
+
+                    if trim_index < 0:
+                        p_as_string_trimmed = '0'
+                    else:
+                        p_as_string_trimmed = p_as_string[:trim_index]
+
+                    percentiles_indexes.append(p_as_string_trimmed + '%')
+            else:
+                sorted_percentiles = [0.25, 0.5, 0.75]
+                percentiles_indexes = ['25%', '50%', '75%']
+
+            index_strings = ['count', 'mean', 'std', 'min']
+            index_strings.extend(percentiles_indexes)
+            index_strings.append('max')
+
+            total_index_chars = numpy.sum(numpy.array([len(s) for s in index_strings]))
+            result_index = sdc.str_arr_ext.pre_alloc_string_array(len(index_strings), total_index_chars)
+            sdc.str_arr_ext.cp_str_list_to_array(result_index, index_strings)
+
+            values = []
+            values.append(numpy.float64(self.count()))
+            values.append(self.mean())
+            values.append(self.std())
+            values.append(self.min())
+            for p in sorted_percentiles:
+                values.append(self.quantile(p))
+            values.append(self.max())
+
+            return pandas.Series(values, result_index)
+
+        return hpat_pandas_series_describe_numeric_impl
+
+    elif isinstance(self.dtype, types.UnicodeType):
+        def hpat_pandas_series_describe_string_impl(self, percentiles=None, include=None, exclude=None):
+
+            objcounts = self.value_counts()
+
+            # temporary workaround for problem in Series.count not skipping Nones
+            # TODO: delete this and use self.count() in append below when test_series_count_string_with_none is fixed
+            dropna_series = self.dropna()
```

Comment:
I think it is better to remove this workaround and skip related tests. Otherwise there is a big chance, we will forget to remove it

Reply:
Done